### PR TITLE
Sequence diagram execution

### PR DESCRIPTION
**Pull request by jtamas96**
Related to #560.

Contains:

- Sequence diagram execution from ui.

- Execution can be cancelled by the user.

- Result of sequence diagram execution is written to console.

### CODE REVIEW

**Comment by djnemeth** [comment]
> I think these UI-related changes do not belong to `api.model.execution`.

Agreed.

> If we cannot find a better location, I propose `export.plantuml` instead.

I don't think there's a better location. We might consider renaming the project.

Additionally, instead of writing directly to the console, couldn't we use our [_Logger_](https://github.com/ELTE-Soft/txtUML/blob/94c2efaf648f9b3a196fc334411946ef3949f1e4/dev/plugins/hu.elte.txtuml.utils/src/hu/elte/txtuml/utils/Logger.java#L6)?

**Comment by jtamas96** [comment]
> Additionally, instead of writing directly to the console, couldn't we use our Logger?

The idea is to give the user a direct response about the results of the last run. I think the console view is a better option for this purpose because we can isolate the results in their own console.

> We might consider renaming the project.

Should I name it to seqDiag (or something like that), or just move the solution and leave the naming as it is for now?

**Comment by djnemeth** [comment]
I suggest the name `seqdiag`. However, with @szokolai-mate we're considering moving Eclipse-dependent control code from existing modules to a dedicated project. I believe this would apply here as well, but I'm generally concerned about separating process control logic from their respective components. @jtamas96 and @kmozsi, what do you think?

**Comment by jtamas96** [comment]
I think in some cases moving the Eclipse-dependent control code would be very useful. For example currently 4 projects are related to the txtUML menu, so we have to make a menu contribution extension in each of the 4 projects in order to add a command to our txtUML menu. I think this is not so readable and I couldn't find a way to determine the order of the commands in the menu, but defining the whole menu with it's commands in one project would make the ordering possible.

**Comment by kmozsi** [comment]
> I suggest the name `seqdiag`.

I like it. 👍 

>  I'm generally concerned about separating process control logic from their respective components.

As process control logic does not strongly belong to its current context, I think the refact is a good idea, and also (eclipse) dependencies would be rationalized by doing so. 

For now, I think the project should be renamed to `seqdiag`, and please move your changes there. Make sure to separate these improvements well from PlantUML export.

**Comment by kmozsi** [comment]
Jenkins, retest this, please.

File: dev/plugins/hu.elte.txtuml.api.model.execution/build.properties [hunk 1]
```diff
@@ -1,4 +1,5 @@
 source.. = src/
 output.. = bin/
 bin.includes = META-INF/,\
-               .
+               .,\
+               plugin.xml
```
Comment (kmozsi) [inline dev/plugins/hu.elte.txtuml.api.model.execution/build.properties]: Please also remove this.

**Comment by djnemeth** [comment]
> As process control logic does not strongly belong to its current context, I think the refact is a good idea, and also (eclipse) dependencies would be rationalized by doing so.

On the other hand, centralizing control logic in a separate project would make the whole feature less modular, i.e. we would lose the ability to achieve #81. @kmozsi, @szokolai-mate, @jtamas96, what do you think?

**Comment by djnemeth** [comment]
> On the other hand, centralizing control logic in a separate project would make the whole feature less modular, i.e. we would lose the ability to achieve #81. @kmozsi, @szokolai-mate, @jtamas96, what do you think?

I have created a dedicated issue for this problem in the form of #601. Not to be dealt with in 0.7.0.

> 8771a56 Fix context menu

@kmozsi, please don't forget to review these changes as well.

**Comment by kmozsi** [comment]
Jenkins, retest this, please.

**Pull request merged**